src/ResourceGenerator.php

```diff
@@ -94,7 +94,13 @@ public function getStrategies(): array
 
     public function fromArray(array $data, ?string $uri = null): HalResource
     {
-        $resource = new HalResource($data);
+        /** @psalm-suppress MixedArrayAccess */
+        $embedEmptyCollections =
+            $this->hydrators->has('config') &&
+            $this->hydrators->get('config')['mezzio-hal']['embed-empty-collections'] ??
+            false;
```

Comment:
```suggestion
        $embedEmptyCollections =
            $this->hydrators->has('config')
            && $this->hydrators->get('config')['mezzio-hal']['embed-empty-collections'] ?? false;
```

The above makes the order of operations more clear. Additionally, our CS has us put operators on the next line.